crates/sui-framework/packages/dwallet-system/sources/dwallet_2pc_mpc_ecdsa_k1.move

```diff
@@ -240,15 +227,9 @@ module dwallet_system::dwallet_2pc_mpc_ecdsa_k1 {
             centralized_party_public_key_share_decommitment_and_proof
         );
 
-        let dwallet = DWallet {
-            id: object::new(ctx),
-            session_id,
-            dwallet_cap_id,
-            output,
-            public_key,
-        };
+        let dwallet = new_dwallet(session_id, dwallet_cap_id, output, public_key, ctx);
         // Create dwallet + make it immutable.
-        transfer::freeze_object(dwallet);
+        transfer::public_freeze_object(dwallet);
```

Comment:
Please explain this change

Reply:
I've refactored the `DWallet` object into a separate module. As a result, we need to utilize public transfers. Alternatively, I can implement a `freeze` method that uses `transfer::freeze_object`, and call `dwallet_object::freeze` from here. However, I believe it's best to proceed with the current version. [https://docs.sui.io/concepts/transfers]